src/components/wms/AnimatedStreamlineMapboxLayer.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+  <div></div>
+</template>
+
+<script setup lang="ts">
+import { type Layer } from '@deltares/fews-wms-requests'
+import { useMap } from '@studiometa/vue-mapbox-gl'
+import { onMounted, onUnmounted, watch } from 'vue'
+
+import {
+  StreamlineStyle,
+  WMSStreamlineLayer,
+  type WMSStreamlineLayerOptions,
+} from '@/lib/streamlines'
+import { configManager } from '@/services/application-config'
+import { type MapboxLayerOptions } from './AnimatedMapboxLayer.vue'
+import type { MapLayerMouseEvent, MapLayerTouchEvent } from 'mapbox-gl'
+
+type StreamlineLayerOptionsFews = Layer['animatedVectors']
+
+interface Props {
+  layerOptions?: MapboxLayerOptions
+  streamlineOptions?: StreamlineLayerOptionsFews
+}
+const props = defineProps<Props>()
+const emit = defineEmits(['doubleclick'])
+
+const { map } = useMap()
+
+const layerId = 'streamlines'
+let layer: WMSStreamlineLayer | null = null
+
+onMounted(() => {
+  if (!props.layerOptions || !props.streamlineOptions) return
+  const options = mergeOptions(props.layerOptions, props.streamlineOptions)
+
+  // Create and add layer.
+  layer = new WMSStreamlineLayer(layerId, options)
+  map.value.addLayer(layer, 'boundary_country_outline')
+  map.value.on('dblclick', onDoubleClick)
+
+  // Make sure we are at the appropriate time, elevation and color scale range
+  // after the visualiser has been initialised.
+  layer.on('load', () => {
+    if (!layer || !props.layerOptions) return
```

Comment:
Is the `!layer` necessary? How can we do a `layer.on('load', ...)` if `layer` does not exist?

Reply:
We can't, but TypeScript doesn't know that...